test/functional/api/buffer_spec.lua

```diff
@@ -707,4 +707,40 @@ describe('api/buf', function()
       eq({3, 0}, curbuf('get_mark', 'v'))
     end)
   end)
+
+  describe('nvim_buf_set_mark', function()
+    it('works with buffer marks', function()
+      curbuf('set_lines', -1, -1, true, {'a', 'bit of', 'text'})
+      curbuf('set_mark', 'z', 1, 1)
+      eq({1, 1}, curbuf('get_mark', 'z'))
+    end)
+    it('works with file/uppercase marks', function()
+      curbuf('set_lines', -1, -1, true, {'a', 'bit of', 'text'})
+      curbuf('set_mark', 'Z', 3, 1)
+      eq({3, 1}, curbuf('get_mark', 'Z'))
+    end)
+  end)
+
+  describe('nvim_buf_del_mark', function()
+    it('works with buffer marks', function()
+      curbuf('set_lines', -1, -1, true, {'a', 'bit of', 'text'})
+      curbuf('set_mark', 'z', 3, 1)
+      curbuf('del_mark', 'z')
+      eq({0, 0}, curbuf('get_mark', 'z'))
+    end)
+    it('works with file/uppercae marks', function()
+      curbuf('set_lines', -1, -1, true, {'a', 'bit of', 'text'})
+      curbuf('set_mark', 'Z', 3, 3)
+      curbuf('del_mark', 'Z')
+      eq({0, 0}, curbuf('get_mark', 'Z'))
+    end)
+    it('won\'t work in file/uppercae marks that do not belong to the buffer',
+      function()
+        local abuf = meths.create_buf(false,true)
+        buffer('set_lines', abuf, -1, -1, true, {'a', 'bit of', 'text'})
+        buffer('set_mark', abuf, 'A', 2, 2)
+        curbuf('del_mark', 'A')
```

Comment:
if del_mark returns a success/fail result, the test becomes much clearer. that's a signal that the silent failure is probably not what we want.
```suggestion
        eq(false, curbuf('del_mark', 'A'))
```